docusaurus/docs/Android/03-ui/06-guides/08-adding-custom-attachments.mdx

```diff
@@ -0,0 +1,169 @@
+# Adding Custom Attachments
+
+## Introduction
+
+By default Stream Chat supports several built-in attachment types like image, video, file and giphy. You can also add your own types of attachments such as location, contact, audio, sticker, etc.
+
+In this guide, we'll demonstrate how to build a date sharing feature. Chat users will be able to pick a date from a calendar, preview their selection in the message input and see the sent attachment within a message in the message list.
+
+This involves doing the following steps:
+1. Customizing the message input in a way that it is capable of sending messages with `date` attachments.
+2. Adding support for `date` attachments in the message list.
+
+:::note
+In this guide, we'll show only the main points concerning custom attachments. Smaller parts will be omitted for the sake of being concise.
+
+You can find the full code from this guide on [GitHub](https://github.com/GetStream/Android-Samples/tree/main/custom-attachments).
+:::
+
+## Sending Date Attachments
+
+To begin with, you'll need to customize [MessageInputView](../04-message-components/04-message-input.mdx) so that the attachment button allows the user to pick dates:
```

Comment:
I think Kanat's version makes slightly more sense, maybe we can even drop the `of all`.
Here's Kanat's version as a suggestion for easier adoption.

```suggestion
First of all, you'll need to customize [MessageInputView](../04-message-components/04-message-input.mdx) to let a user pick a date through the attachment button:
```

Reply:
Agreed but I'd use this:

`First of all, you'll need to customize MessageInputView to let the user pick a date using the attachment button:`